<p>I think you just want to catch it in a catch, and then rethrow it or another exception with whatever values you want as parameters.  Why do you want to do this with a finally?</p>